app/models/root-store/root-store.ts

```diff
@@ -11,15 +11,28 @@ import { UserModel } from "../user/user"
  * A RootStore model.
  */
 // prettier-ignore
-export const RootStoreModel = types.model("RootStore").props({
-  routePlan: types.optional(RoutePlanModel, {} as any),
-  trainRoutes: types.optional(trainRoutesModel, {} as any),
-  recentSearches: types.optional(RecentSearchesModel, {} as any),
-  favoriteRoutes: types.optional(FavoritesModel, {} as any),
-  settings: types.optional(SettingsModel, {} as any),
-  ride: types.optional(RideModel, {} as any),
-  user: types.optional(UserModel, {} as any)
-})
+export const RootStoreModel = types
+  .model("RootStore")
+  .props({
+    routePlan: types.optional(RoutePlanModel, {} as any),
+    trainRoutes: types.optional(trainRoutesModel, {} as any),
+    recentSearches: types.optional(RecentSearchesModel, {} as any),
+    favoriteRoutes: types.optional(FavoritesModel, {} as any),
+    settings: types.optional(SettingsModel, {} as any),
+    ride: types.optional(RideModel, {} as any),
+    user: types.optional(UserModel, {} as any),
+  })
+  .actions((self) => ({
+    clearAllData() {
+      self.routePlan = RoutePlanModel.create({})
+      self.trainRoutes = trainRoutesModel.create({})
+      self.recentSearches = RecentSearchesModel.create({})
+      self.favoriteRoutes = FavoritesModel.create({})
+      self.settings = SettingsModel.create({})
+      self.ride = RideModel.create({})
+      self.user = UserModel.create({})
+    },
```

Comment:
MobX offers the [`applySnapshot`](https://mobx-state-tree.js.org/API/#applysnapshot) method, which achieves the same result in a cleaner and more efficient way :)

```suggestion
    clearAllData() {
      applySnapshot(self, {
        routePlan: {},
        trainRoutes: {},
        recentSearches: {},
        favoriteRoutes: {},
        settings: {},
        ride: {},
        user: {}
      })
    },```